frontend/src/actions/InlineTabActions.js

```diff
@@ -0,0 +1,223 @@
+ */
+export function updateInlineTabItemFields({ inlineTabId, fieldsByName }) {
+  return {
+    type: UPDATE_INLINE_TAB_ITEM_FIELDS,
+    payload: { inlineTabId, fieldsByName },
+  };
+}
+
+/*
+ * @method updateInlineTabWrapperFields
+ * @summary Action creator for updating the fields for the `InlineTab` Wrapper
+ *
+ * @param {string} inlineTabWrapperId
+ * @param {string} rowId
+ * @param {object} fieldsByName
+ */
+export function updateInlineTabWrapperFields({
+  inlineTabWrapperId,
+  rowId,
+  response,
+}) {
+  return {
+    type: UPDATE_INLINE_TAB_WRAPPER_FIELDS,
+    payload: { inlineTabWrapperId, rowId, response },
+  };
+}
+
+/*
+ * Action creator called to set the inlineTabWrapper branch in the redux store with the data payload
+ */
+export function setInlineTabWrapperData({ inlineTabWrapperId, data }) {
+  return {
+    type: SET_INLINE_TAB_WRAPPER_DATA,
+    payload: { inlineTabWrapperId, data },
+  };
+}
+
+/*
+ * Action creator called to set the showMore value for the corresponding inlineTabWrapperId
+ */
+export function setInlineTabShowMore({ inlineTabWrapperId, showMore }) {
+  return {
+    type: SET_INLINE_TAB_SHOW_MORE,
+    payload: { inlineTabWrapperId, showMore },
+  };
+}
+
+/*
+ * Action creator called to set the inlineTab branch in the redux store with the data payload
+ */
+export function setInlineTabLayoutAndData({ inlineTabId, data }) {
+  return {
+    type: SET_INLINE_TAB_LAYOUT_AND_DATA,
+    payload: { inlineTabId, data },
+  };
+}
+
+/*
+ * Action creator called to set the inlineTab AddNew form related data in the store, visible is for toggling the visibility
+ */
+export function setInlineTabAddNew({ visible, windowId, tabId, rowId, docId }) {
+  return {
+    type: SET_INLINE_TAB_ADD_NEW,
+    payload: { visible, windowId, tabId, rowId, docId },
+  };
+}
+
+/*
+ * @method fetchInlineTabWrapperData
+ * @summary Action creator for fetching the data for the `InlineTab` Wrapper (note: wrapper not the inline tab item!)
+ *
+ * @param {string} windowId
+ * @param {string} tabId
+ * @param {string} docId
+ * @param {string} query
+ */
+export function fetchInlineTabWrapperData({
+  windowId,
+  tabId,
+  docId,
+  query,
+  rowId,
+  postDeletion,
+}) {
+  return (dispatch) => {
+    dispatch(fetchTab({ tabId, windowId, docId, query })).then((tabData) => {
+      /** - if we have the rowId it means we have a new record addition, so we put that at the end of the array - only if this doesn't happen as a result of deletion */
+      if (rowId && !postDeletion) {
+        const lastAdditionIndex = tabData.findIndex(
+          (item) => item.rowId === rowId
+        );
+        if (lastAdditionIndex) {
+          const tempData = tabData[lastAdditionIndex];
+          tabData.splice(lastAdditionIndex, 1);
+          tabData.splice(tabData.length, 0, tempData);
+        }
+        dispatch(
+          setInlineTabShowMore({
+            inlineTabWrapperId: `${windowId}_${tabId}_${docId}`,
```

Comment:
calling each time a function for that would be counter performance. But a solution would be to just define the selector in one place instead of how I put it now. This Is fixed. Ty